src/components/FileDropZone.tsx

```diff
@@ -14,54 +23,119 @@ function convertToPlainText(html: string): string {
 }
 /* eslint-enable functional/immutable-data */
 
-function createSourceFromParagraphs(
-  paragraphs: Array<string>,
+function createRelationsFromParagraphNodes(
+  nodes: KnowNode[],
   myself: PublicKey
-): {
-  nodes: KnowNode[];
-  relations: Relations;
-  topNodeID: LongID;
-} {
-  const topParagraph = paragraphs[0];
-  const furtherParagraphs = paragraphs.slice(1);
+): [relations: Relations, topNodeID: LongID] {
+  const topParagraph = nodes[0];
+  const furtherParagraphs = nodes.slice(1);
+  const relations = bulkAddRelations(
+    newRelations(topParagraph.id, "", myself),
+    furtherParagraphs.map((n) => n.id)
+  );
+  return [relations, topParagraph.id];
+}
 
-  const topNode = newNode(topParagraph, myself);
-  const innerNodes = furtherParagraphs.map((paragraph) => {
-    return newNode(paragraph, myself);
+export function createNodesFromMarkdown(
+  markdown: string,
+  baseID: string,
+  myself: PublicKey
+): KnowNode[] {
+  const markdownParagraphs = markdown.split("\n\n");
+  const plainTextParagraphs = markdownParagraphs.map((paragraph: string) => {
+    const md = new MarkdownIt();
+    return convertToPlainText(md.render(paragraph));
```

Comment:
don't we dangerously set the innerHtml at convertToPlainText. Might be an attack vector...?

Reply:
we don't render it though. We just create the html element using the markdown parser and then extract the text.